tests/common/dualtor/dualtor_utils.py

```diff
@@ -112,3 +173,44 @@ def recover_all_directions(mux_server_url, physical_port):
     data = {"out_ports": [TOR_A, TOR_B, NIC]}
     pytest_assert(_post(server_url, data), "Failed to set output on all directions")
 
+def check_simulator_read_side(mux_server_url, physical_port):
+    """
+    Retrieve the current active tor from y_cable simulator server.
+    Args:
+        mux_server_url: a str, the address of mux server
+        physical_port: physical port on switch, an integer starting from 1
+    Returns:
+        1 if TOR_A is active
+        2 if TOR_B is active
+        -1 for exception or inconstient status
+    """
+    server_url = _url(mux_server_url, physical_port)
+    res = _get(server_url)
+    if not res:
+        return -1
+    active_side = res["active_side"]
+    if active_side == TOR_A:
+        return 1
+    elif active_side == TOR_B:
+        return 2
+    else:
+        return -1
+
+@pytest.fixture(scope='module')
+def toggle_all_simulator_ports_to_tor_a(mux_server_url):
+    """
+    A module level fixture to toggle all ports to tor_a
+    """
+    server_url = _url(mux_server_url)
+    data = {"active_side": TOR_A}
+    pytest_assert(_post(server_url, data), "Failed to toggle all ports to TOR A")
+
+@pytest.fixture(scope='module')
+def toggle_all_simulator_ports_to_tor_b(mux_server_url):
+    """
+    A module level fixture to toggle all ports to tor_b
+    """
+    server_url = _url(mux_server_url)
+    data = {"active_side": TOR_B}
+    pytest_assert(_post(server_url, data), "Failed to toggle all ports to TOR A")
```

Comment:
I think this function is dependent on PR https://github.com/Azure/sonic-mgmt/pull/2703.

The simulator server also supports two other options:
* "toggle": Just toggle the active side for all ports
* "random": Randomly set active side for each of the port.

Can you consider adding fixtures for these two options as well?